core/chains/evm/forwarders/forwarder_manager.go

```diff
@@ -0,0 +1,291 @@
+
+type FwdMgr struct {
+	utils.StartStopOnce
+	ORM       ORM
+	evmClient evmclient.Client
+	logger    logger.SugaredLogger
+	logpoller *evmlogpoller.LogPoller
+
+	// TODO(samhassan): sendersCache should be an LRU capped cache
+	sendersCache map[common.Address][]common.Address
+	latestBlock  int64
+
+	authRcvr    authorized_receiver.AuthorizedReceiverInterface
+	offchainAgg offchain_aggregator_wrapper.OffchainAggregatorInterface
+
+	ctx    context.Context
+	cancel context.CancelFunc
+
+	cacheMu sync.RWMutex
+	chStop  chan struct{}
+	wg      sync.WaitGroup
+}
+
+func NewFwdMgr(db *sqlx.DB, client evmclient.Client, logpoller *evmlogpoller.LogPoller, l logger.Logger, cfg pg.LogConfig) *FwdMgr {
+	lggr := logger.Sugared(l.Named("EVMForwarderManager"))
+	fwdMgr := FwdMgr{
+		logger:       lggr,
+		evmClient:    client,
+		ORM:          NewORM(db, lggr, cfg),
+		logpoller:    logpoller,
+		sendersCache: make(map[common.Address][]common.Address),
+		chStop:       make(chan struct{}),
+		cacheMu:      sync.RWMutex{},
+		wg:           sync.WaitGroup{},
+		latestBlock:  0,
+	}
+	return &fwdMgr
+}
+
+// Start starts Forwarder Manager.
+func (f *FwdMgr) Start() error {
+	return f.StartOnce("EVMForwarderManager", func() error {
+		f.logger.Debug("Initializing EVM forwarder manager")
+
+		fwdrs, err := f.ORM.FindForwardersByChain(utils.Big(*f.evmClient.ChainID()))
+		if err != nil {
+			return errors.Wrapf(err, "Failed to retrieve forwarders for chain %d", f.evmClient.ChainID())
+		}
+		f.ctx, f.cancel = context.WithCancel(context.Background())
+		if len(fwdrs) != 0 {
+			f.initForwardersCache(f.ctx, fwdrs)
+			f.subscribeForwardersLogs(fwdrs)
+		}
+
+		f.authRcvr, err = authorized_receiver.NewAuthorizedReceiver(common.Address{}, f.evmClient)
+		if err != nil {
+			return errors.Wrap(err, "Failed to init AuthorizedReceiver")
+		}
+
+		f.offchainAgg, err = offchain_aggregator_wrapper.NewOffchainAggregator(common.Address{}, f.evmClient)
+		if err != nil {
+			return errors.Wrap(err, "Failed to init OffchainAggregator")
+		}
+
+		f.wg.Add(1)
+		go f.runLoop()
+		return nil
+	})
+}
+
+// TODO(samhassan): this should be aware of job type to decide how to fetch senders list.
+func (f *FwdMgr) MaybeForwardTransaction(from common.Address, to common.Address, encodedPayload []byte) (fwdAddr common.Address, fwdPayload []byte, err error) {
+
+	senders, err := f.getContractSenders(to)
+	if err != nil {
+		return to, encodedPayload, errors.Wrap(err, "Skipping forwarding transaction")
+	}
+
+	// Gets current forwarders that are in `to` senders
+	fwdrs, err := f.ORM.FindForwardersInListByChain(utils.Big(*f.evmClient.ChainID()), senders)
+	if err != nil {
+		return to, encodedPayload, errors.Wrap(err, "Skipping forwarding transaction")
+	}
+
+	for _, fwdr := range fwdrs {
+		eoas, err := f.getContractSenders(fwdr.Address)
+		if err != nil {
+			f.logger.Warnf("Failed to get forwarder senders %s", err)
```

Comment:
Is it really safe to warn and continue here? Wouldn't this mean we might not get the right account to forward to, and fail the transaction?

Reply:
I agree with bumping this to an error but I think we should gracefully continue otherwise one corrupt add forwarder operation will break all forwarding logic. (bumped level)